src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseCachingService.kt

```diff
@@ -55,17 +78,29 @@ class TestCaseCachingService {
         private val lines = mutableMapOf<Int, LineTestCaseCache>()
         private val linesLock = Object()
 
+        // Used for retrieving references of unique test cases
+        private val caseIndex = mutableMapOf<String, CachedCompactTestCase>()
+        private val caseIndexLock = Object()
+
         /**
          * Insert test cases into the file cache.
          *
          * @param report the report containing the test cases
          */
         fun putIntoCache(report: CompactReport) {
             report.testCaseList.values.forEach { testCase ->
+                val cachedCompactTestCase = CachedCompactTestCase.fromCompactTestCase(testCase, this)
+
+                synchronized(caseIndexLock) {
```

Comment:
synchronized, Why was that choice made?

Reply:
That's Marti's code, but this is for thread-safety, i guess

Reply:
Yep